core/src/main/java/io/grpc/internal/ServerImpl.java

```diff
@@ -326,9 +334,20 @@ public void transportTerminated() {
     }
 
     @Override
+    public StatsTraceContext methodDetermined(String methodName, Metadata headers) {
+      return StatsTraceContext.newServerContext(
+          methodName, censusFactory, headers, stopwatchSupplier);
+    }
+
+    @Override
     public ServerStreamListener streamCreated(final ServerStream stream, final String methodName,
         final Metadata headers) {
```

Comment:
No, I wasn't suggesting getting rid of `methodDetermined()`. Instead, keep `methodDetermined()` but instead of implicitly passing `StatusTraceContext` via the stream, pass it explicitly as an argument.